src/plugins/interactive-media/interactive-media-editor.js

```diff
@@ -1,102 +1,111 @@
 import { sectionEditorProps } from '../../ui/default-prop-types.js';
 import StorageFilePicker from '../../components/storage-file-picker.js';
 import ObjectMaxWidthSlider from '../../components/object-max-width-slider.js';
 
+const ReactPlayer = reactPlayerNs.default || reactPlayerNs;
+
 const FormItem = Form.Item;
 const RadioGroup = Radio.Group;
 const TextArea = Input.TextArea;
 const RadioButton = Radio.Button;
 
 const videoTypes = ['mp4', 'mov', 'avi', 'mkv'];
 const audioTypes = ['mp3', 'flac', 'aac', 'wav'];
+const supportedAspectRatios = [{ h: 16, v: 9 }, { h: 4, v: 3 }];
 
-const ensurePartsOrder = parts => parts.sort(by(part => part.startTimecode));
-const getAspectRatioText = aspectRatio => `${aspectRatio.h}:${aspectRatio.v}`;
-
-const getMediaType = path => {
-  const sanitizedPath = (path || '').trim();
-  const extensionMatches = sanitizedPath.match(/\.([0-9a-z]+)$/i);
-  const extension = extensionMatches && extensionMatches[1];
-
-  if (!extension) {
-    return MEDIA_TYPE.none;
-  }
-  if (audioTypes.includes(extension)) {
-    return MEDIA_TYPE.audio;
-  }
-  if (videoTypes.includes(extension)) {
-    return MEDIA_TYPE.video;
-  }
-  return MEDIA_TYPE.unknown;
+const formItemLayout = {
+  labelCol: { span: 4 },
+  wrapperCol: { span: 14 }
 };
 
 function InteractiveMediaEditor({ content, onContentChanged, publicStorage, privateStorage }) {
   const clientConfig = useService(ClientConfig);
+  const hiddenPlayerContainerRef = useRef(null);
   const { t } = useTranslation('interactiveMedia');
 
-  const { sourceType, sourceUrl, text, width, aspectRatio, showVideo } = content;
+  const getAspectRatioText = givenAspectRatio => `${givenAspectRatio.h}:${givenAspectRatio.v}`;
+  const ensureChaptersOrder = chapters => chapters.sort(by(chapter => chapter.startTimecode));
+  const getDefaultChapter = () => ({ key: uniqueId.create(), title: t('defaultChapterTitle'), startTimecode: 0 });
 
-  const supportedAspectRatios = [{ h: 16, v: 9 }, { h: 4, v: 3 }];
   const defaultAspectRatio = supportedAspectRatios[0];
+  const [isDeterminingDuration, setIsDeterminingDuration] = useState(false);
+  const { sourceType, sourceUrl, sourceDuration, chapters, text, width, aspectRatio, showVideo } = content;
 
-  const formItemLayout = {
-    labelCol: { span: 4 },
-    wrapperCol: { span: 14 }
-  };
-
-  const lastTimecode = 7;
-  const initialParts = [
-    {
-      title: 'the brown fox',
-      startTimecode: 0
-    },
-    {
-      title: 'jumped the veeeeeeeeeery looooong fence',
-      startTimecode: 2 * 1000
-    },
-    {
-      title: 'the end',
-      startTimecode: lastTimecode * 1000
+  const getMediaType = path => {
+    const sanitizedPath = (path || '').trim();
+    const extensionMatches = sanitizedPath.match(/\.([0-9a-z]+)$/i);
+    const extension = extensionMatches && extensionMatches[1];
+
+    if (!extension) {
+      return MEDIA_TYPE.none;
+    }
+    if (audioTypes.includes(extension)) {
+      return MEDIA_TYPE.audio;
```

Comment:
I know this was there before, but I just realized we have (pretty much) the same code in `github-flavored-markdown.js`. Maybe you wanna check if we could easily unify those two places (there are a few more formats there, but they all should work)?

Reply:
Nice, I didn't know about this other place, thanks 👍 
Unified the media type identification.